src/fides/api/main.py

```diff
@@ -207,7 +209,10 @@ async def setup_server() -> None:
     await configure_db(CONFIG.database.sync_database_uri)
 
     logger.info("Validating SaaS connector templates...")
-    load_registry(registry_file)
+    registry = load_registry(registry_file)
+    db = get_api_session()
+    update_saas_configs(registry, db)
+    db.close()
```

Comment:
What if this fails?

Reply:
good point - i can certainly wrap this in a try/except, as it had been before, so that we can continue booting up without problem. i think it makes sense to do that for at least the db session/update block, since that shouldn't be necessary for the server to function.

i'm a little bit unsure if we want to catch errors on `load_registry`. if that fails for whatever reason, i think it may not make sense for the server to continue booting up, because it's debatably in a state where it shouldn't be used. that being said, i was accidentally triggering this behavior on unified fides in some testing, and the error was not well reported at all - i couldn't find any trace of it, but the webserver just hung and refused to start up. i think that's a separate issue -- something along the lines of properly bubbling up errors from this startup hook -- but i did want to get that on the table here for discussion.

Reply:
OK I agree with you then that we should not ignore these errors, that we should prevent the server from starting, so I'll go ahead and approve this work.  

Maybe we file a second issue for this latter point you mention? Generally it's feeling like server issues are harder to troubleshoot over here than in fidesops. I haven't tested the startup issues, but I've seen 500 internal server errors on endpoints just get swallowed with no indication in the logs of the request.

Reply:
seeing that there is a `db.close()` here, as a best practice I think this should also use a context manager to prevent leaving any dangling resources (https://book.pythontips.com/en/latest/context_managers.html#context-managers)

Reply:
thanks for that @ThomasLaPiana, will push an update up here now.

and agreed @pattisdr , let me do a bit more research on some of the server issues and i'll open up a new issue